packages/amplify-e2e-core/src/utils/headless.ts

```diff
@@ -75,13 +75,15 @@ const executeHeadlessCommand = async (
   request: AnyHeadlessRequest,
   reject: boolean = true,
   allowDestructiveUpdates: boolean = false,
-  settings?: any,
-): Promise<ExecaChildProcess<String>> => {
+  settings:any = {testingWithLatestCodebase : true },
+): Promise<any> => {
   const args = [operation, category, '--headless'];
   if (allowDestructiveUpdates) {
     args.push('--allow-destructive-graphql-schema-updates');
   }
-  return await execa(getCLIPath(settings.testingWithLatestCodebase), args, { input: JSON.stringify(request), cwd, reject });
+  const isCodeBaseLatest = settings.testingWithLatestCodebase;
+  const cliPath = getCLIPath(isCodeBaseLatest);
+  return await execa(cliPath, args, { input: JSON.stringify(request), cwd, reject });
```

Comment:
I think having two variable assignments here is overkill and doesn't really add to the readability. Can we keep the return type as it was?

```suggestion
  settings: any = { testingWithLatestCodebase : true },
): Promise<ExecaChildProcess<String>> => {
  const args = [operation, category, '--headless'];
  if (allowDestructiveUpdates) {
    args.push('--allow-destructive-graphql-schema-updates');
  }
  const cliPath = getCLIPath(settings.testingWithLatestCodebase);
  return await execa(cliPath, args, { input: JSON.stringify(request), cwd, reject });
```